src/cargo/core/compiler/context/mod.rs

```diff
@@ -150,7 +168,8 @@ impl<'a, 'cfg> Context<'a, 'cfg> {
             files: None,
         };
 
-        cx.probe_target_info()?;
+        cx.compilation.host_dylib_path = cx.host_info.sysroot_libdir.clone();
+        cx.compilation.target_dylib_path = cx.target_info.sysroot_libdir.clone();
```

Comment:
Should we move this into `Compilation::new` as well?

Reply:
I definitely thought about it. I decided against it for now mostly on style (passing around long names is kind of ugly). I just looked into how the rest of `Compilation` gets initialized, and it looks like pretty much everything gets initialized after the fact from throughout `Context`, so passing these two in at `new()` is not a win for consistency. What do you think?

Reply:
Yeah, let's leave it as is